src/tools/pdf_files.py

```diff
@@ -1,7 +1,32 @@
 import langchain_core.tools as lcct
+import fitz
+import unicodedata
+import os
 
 
 @lcct.tool
-def parse_pdf(file: str) -> str:
-    """Mock: Return fake parsed content from PDF."""
-    return "Parsed content from the PDF (mocked)."
+def parse_pdf(file_path: str) -> str:
+    """
+    Extracts and returns the full text content from a PDF file using PyMuPDF.
+
+    Parameters:
+    - file_path (str): Path to the PDF file.
+
+    Returns:
+    - str: Extracted text from the PDF.
+    """
+    if not os.path.exists(file_path):
+        raise FileNotFoundError(f"The file at {file_path} does not exist.")
+
+    try:
```

Comment:
don't use `try`, just throw an error, I believe `@tool` decorator handles it.

Reply:
I can't just raise an error without try maybe I'll just remove the try except as a whole

Reply:
that is what I meant. Don't use try at all.